tests/platform/broadcom/files/ser_injector.py

```diff
@@ -0,0 +1,306 @@
+        print("SER Test timed out for memories (%s): %s" % (len(self.mem_verification_pending), self.mem_verification_pending))
+        print("SER Test is not supported for memories (%s): %s" % (len(self.mem_ser_unsupported), self.mem_ser_unsupported))
+
+    def enable_memory_scan(self, cmd, interval_usec, rate):
+        '''
+        @summary: Enable Broadcom memory scan
+        @param cmd: Broadcom to use
+        @param interval_usec: memory scanner interval i usec
+        @param rate: rate (number of entries) per interval
+        '''
+        count = 0
+        while count < 3:
+            count += 1
+            stdout, stderr = run_cmd(["bcmcmd", cmd + " interval=" + str(interval_usec) + " rate=" + str(rate)])
+            lines = stdout.decode("utf-8").split("\n")
+            if lines[1].find('mSCAN: Started on unit 0') > -1:
+                return
+
+        raise ValueError('Failed to start memory scanner: %s' % cmd) 
+
+    def enable_mem_scan(self, interval_usec, rate):
+        '''
+        @summary: Wrapper around enable_memory_scan
+        @param interval_usec: memory scanner interval i usec
+        @param rate: rate (number of entries) per interval
+        '''
+        self.enable_memory_scan('memscan', interval_usec, rate)
+
+    def enable_sram_scan(self, interval_usec, rate):
+        '''
+        @summary: Enable Broadcom sram scan
+        @param interval_usec: memory scanner interval i usec
+        @param rate: rate (number of entries) per interval
+        '''
+        self.enable_memory_scan('sramscan', interval_usec, rate)
+
+    def verify_ser(self, entry, log):
+        '''
+        @summary: verify SER log entry
+        @param entry: indext of the memory table where SER was injected
+        @param log: syslog log line
+
+        @return: memory table name
+        @return: Flag if SER injection entry matches log line entry
+        '''
+        m = re.search("^.*addr:(.*) port.*index: (\d+)", log)
+
+        address = int(m.group(1), 16)
+        mem_entry = int(m.group(2))
+
+        memory = self.bcmMemory.get_memory_by_address()
+        if address in memory:
+            m = memory[address]
+            return memory[address], entry == mem_entry
+
+        return None, None
+
+    def inject_ser(self, mem, index = 0):
+        '''
+        @summary: Inject SER error suing Broadcom ser inject command
+        @param mem: name of the memory table to inject SER into
+        @param index: index of the entry to inject SER into
+        '''
+        return run_cmd(["bcmcmd",  "ser inject memory=" + mem + " index=" + str(index)])
+
+    def verify_and_update_test_result(self, entry, line):
+        '''
+        @summary: Verify log line and update test result
+        @param entry: index of the entry to inject SER into
+        @param log: syslog log line
+        '''
+        mem, f = self.verify_ser(entry, line)
```

Comment:
what is `f` here? Can you please make longer variable name?

Reply:
replaced it with descriptive var name